src/Pager.js

```diff
@@ -0,0 +1,282 @@
+  _isSwiping = new Value(FALSE);
+
+  // Whether the update was due to swipe gesture
+  // Remember to set it when transition needs to occur
+  _isSwipeGesture = new Value(FALSE);
+
+  _clock = new Clock();
+
+  _routesLength = new Value(this.props.navigationState.routes.length);
+
+  // tslint:disable-next-line: strict-boolean-expressions
+  _layoutWidth = new Value(this.props.layout.width || 320);
+
+  _swipeDistanceThreshold = divide(this._layoutWidth, 1.75);
+  _swipeVelocityThreshold = 1200;
+
+  transitionTo = (index: Animated.Node<number>) => {
+    const state = {
+      position: this._position,
+      velocity: this._velocityX,
+      time: new Value(0),
+      finished: new Value(0),
+    };
+
+    const config = {
+      ...SPRING_CONFIG,
+      toValue: new Value(0),
+    };
+
+    return block([
+      cond(clockRunning(this._clock), NOOP, [
+        // Animation wasn't running before
+        // Set the initial values and start the clock
+        set(config.toValue, multiply(index, this._layoutWidth, -1)),
+        set(state.finished, 0),
+        set(state.time, 0),
+        set(this._index, index),
+        startClock(this._clock),
+      ]),
+      // Animate the values with a spring
+      spring(this._clock, state, config),
+      cond(state.finished, [
+        // When spring animation finishes, stop the clock
+        stopClock(this._clock),
+        // Reset gesture and velocity from previous gesture
+        set(this._gestureX, 0),
+        set(this._velocityX, 0),
+        call([this._index], ([value]) => {
+          // If the index changed, and previous spring was finished, update state
+          const route = this.props.navigationState.routes[Math.round(value)];
+
+          this.props.jumpTo(route.key);
+        }),
+      ]),
+    ]);
+  };
+
+  handleGestureEvent = event([
+    {
+      nativeEvent: {
+        translationX: this._gestureX,
+        velocityX: this._velocityX,
+        state: this._gestureState,
+      },
+    },
+  ]);
+
+  translateX = block([
+    onChange(
+      // Index changed from outside
+      this._nextIndex,
+      cond(or(eq(this._index, this._nextIndex), this._isSwipeGesture), NOOP, [
+        // Stop any running animations
+        stopClock(this._clock),
+        // Update the index to trigger the transition
+        set(this._index, this._nextIndex),
+      ])
+    ),
+    cond(
+      eq(this._gestureState, State.ACTIVE),
+      [
+        cond(this._isSwiping, NOOP, [
+          // We weren't dragging before, set it to true
+          set(this._isSwiping, TRUE),
+          set(this._isSwipeGesture, TRUE),
+          // Also update the drag offset to the last position
+          set(this._offsetX, this._position),
+        ]),
+        // Update position with previous offset + gesture distance
+        set(this._position, add(this._offsetX, this._gestureX)),
+        // Stop animations while we're dragging
+        stopClock(this._clock),
```

Comment:
I think it could be `cond(clockRunning(this._clock), stopClock(this._clock))` I think it will be clearer then what's the purpose of this line.